core/src/main/java/com/gentics/mesh/cli/AbstractBootstrapInitializer.java

```diff
@@ -944,7 +949,60 @@ protected void initLanguageSet(LanguageSet languageSet) {
 	 * @param vertxOptions
 	 *            Vert.x options
 	 */
-	protected abstract Vertx createClusteredVertx(MeshOptions options, VertxOptions vertxOptions);
+	protected Vertx createClusteredVertx(MeshOptions options, VertxOptions vertxOptions) {
+		clusterManager = db.clusterManager().getVertxClusterManager();
+		vertxOptions.setClusterManager(clusterManager);
+		String localIp = options.getClusterOptions().getNetworkHost();
+
+		Integer clusterPort = options.getClusterOptions().getVertxPort();
+		int vertxClusterPort = clusterPort == null ? 0 : clusterPort;
+
+		EventBusOptions eventbus = vertxOptions.getEventBusOptions();
+		eventbus.setHost(localIp);
+		eventbus.setPort(vertxClusterPort);
+		eventbus.setClusterPublicHost(localIp);
+		eventbus.setClusterPublicPort(vertxClusterPort);
+
+		if (log.isDebugEnabled()) {
+			log.debug("Using Vert.x cluster port {" + vertxClusterPort + "}");
+			log.debug("Using Vert.x cluster public port {" + vertxClusterPort + "}");
+			log.debug("Binding Vert.x on host {" + localIp + "}");
+		}
+		CompletableFuture<Vertx> fut = new CompletableFuture<>();
+		Vertx.clusteredVertx(vertxOptions, rh -> {
+			log.info("Created clustered Vert.x instance");
+			if (rh.failed()) {
+				Throwable cause = rh.cause();
+				log.error("Failed to create clustered Vert.x instance", cause);
+				fut.completeExceptionally(new RuntimeException("Error while creating clusterd Vert.x instance", cause));
+				return;
+			}
+			Vertx vertx = rh.result();
+			fut.complete(vertx);
+		});
+		try {
+			return fut.get(getClusteredVertxInitializationTimeoutInSeconds(), SECONDS);
+		} catch (Exception e) {
+			throw new RuntimeException("Error while creating clusterd Vert.x instance");
+		}
+	}
+
+	/**
+	 * The timeout used when initializing a clustered vertx instance. If the instance is not available during the provided
+	 * timeout, and exception will be thrown
+	 * @return
+	 */
+	public int getClusteredVertxInitializationTimeoutInSeconds() {
+		return 10;
```

Comment:
How critical is to have this configurable?

Reply:
I would avoid making it user-configurable for now, since it requires more mantainance work. Once we have the final version of mdm, we can evaluate whether we need more fine grained configuration from the user